### PR TITLE
[JENKINS-57678] - Change getInstance() (deprecated) to get()

### PR DESCRIPTION
**Pull request by GustavoBezerra**
See [JENKINS-57678](https://issues.jenkins-ci.org/browse/JENKINS-57678).

### Proposed changelog entries

The method Jenkins.getInstance() is deprecated, and the recommendation is to use the get method instead.

### Submitter checklist

- [x] JIRA issue is well described

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Note that `Jenkins.get()` and `Jenkins.getInstance` have different behavior when Jenkins is starting or tearing down. `getInstance()` return `null`, and it's valid replacement in new APIs would be `getInstanceOrNull()`. On the other hand, `Jenkins.get()` should be used in all cases where there is no null checks. Fortunately all the tricky places have been already refactored, so this PR is fine

Unrelated changes should be reverted

File: core/src/main/java/hudson/PluginManager.java [hunk 1]
```diff
@@ -1521,11 +1521,11 @@ public HttpResponse doInstallPlugins(StaplerRequest req) throws IOException {
     }
 
     private void trackInitialPluginInstall(@Nonnull final List<Future<UpdateCenter.UpdateCenterJob>> installJobs) {
-        final Jenkins jenkins = Jenkins.getInstance();
+        final Jenkins jenkins = Jenkins.get();
         final UpdateCenter updateCenter = jenkins.getUpdateCenter();
         final Authentication currentAuth = Jenkins.getAuthentication();
 
-        if (!Jenkins.getInstance().getInstallState().isSetupComplete()) {
+        if (!Jenkins.get().getInstallState().isSetupComplete()) {
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/hudson/PluginManager.java]: Changes reverted

File: core/src/main/java/hudson/model/BuildStepListener.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
      * Returns all the registered {@link BuildStepListener}s.
      */
     public static ExtensionList<BuildStepListener> all() {
-        // TODO should have a null-safe version when Jenkins.getInstance() is null; would require changes in ExtensionList
+        // TODO should have a null-safe version when Jenkins.get() is null; would require changes in ExtensionList
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/hudson/model/BuildStepListener.java]: Changes reverted

File: core/src/main/java/hudson/security/AuthorizationStrategy.java [hunk 3]
```diff
@@ -233,7 +231,7 @@ public String getDisplayName() {
             }
 
             @Override
-            public @Nonnull AuthorizationStrategy newInstance(StaplerRequest req, JSONObject formData) throws FormException {
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/hudson/security/AuthorizationStrategy.java]: Changes reverted

File: core/src/main/java/jenkins/diagnostics/URICheckEncodingMonitor.java [hunk 4]
```diff
@@ -30,8 +28,8 @@ public String getDisplayName() {
         return Messages.URICheckEncodingMonitor_DisplayName();
     }
 
-    public FormValidation doCheckURIEncoding(StaplerRequest request) throws IOException {
-        Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
+    public FormValidation doCheckURIEncoding(StaplerRequest request) {
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/jenkins/diagnostics/URICheckEncodingMonitor.java]: Changes reverted

File: core/src/main/java/jenkins/management/AdministrativeMonitorsDecorator.java [hunk 5]
```diff
@@ -90,7 +90,7 @@ public int getActiveAdministrativeMonitorsCount() {
 
     /**
      * Whether the administrative monitors notifier should be shown.
-     * @return true iff the administrative monitors notifier should be shown.
+     * @return true if the administrative monitors notifier should be shown.
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/jenkins/management/AdministrativeMonitorsDecorator.java]: Sorry! I've never seen this "iff" before!
Changes reverted.

File: core/src/main/java/jenkins/management/AdministrativeMonitorsDecorator.java [hunk 6]
```diff
@@ -107,19 +107,14 @@ public boolean shouldDisplay() throws IOException, ServletException {
         List<Ancestor> ancestors = req.getAncestors();
 
         if (ancestors == null || ancestors.size() == 0) {
-            // ???
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/jenkins/management/AdministrativeMonitorsDecorator.java]: I have no idea too! But, it's unrelated.
Changes reverted.

File: core/src/main/java/jenkins/tasks/SimpleBuildStep.java [hunk 7]
```diff
@@ -25,34 +25,16 @@
...
-import hudson.tasks.BuildStepDescriptor;
-import hudson.tasks.BuildStepMonitor;
-import hudson.tasks.Builder;
-import hudson.tasks.Publisher;
-import java.io.IOException;
-import java.util.Collection;
-import java.util.LinkedList;
-import java.util.List;
-import javax.annotation.Nonnull;
+import hudson.model.*;
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/jenkins/tasks/SimpleBuildStep.java]: Changes reverted.

File: core/src/main/java/jenkins/tasks/SimpleBuildStep.java [hunk 8]
```diff
@@ -102,32 +84,4 @@ void perform(@Nonnull Run<?,?> run, @Nonnull FilePath workspace, @Nonnull Launch
 
     }
 
-    @SuppressWarnings("rawtypes")
-    @Restricted(NoExternalUse.class)
-    @Extension
-    final class LastBuildActionFactory extends TransientActionFactory<Job> {
```
[the remark this answers is not in the document]
Comment (GustavoBezerra) [reply core/src/main/java/jenkins/tasks/SimpleBuildStep.java]: Changes reverted.

**Comment by GustavoBezerra** [comment]
Changes reverted according  review

File: core/src/main/java/hudson/PluginManager.java [hunk 9]
```diff
@@ -1521,11 +1521,11 @@ public HttpResponse doInstallPlugins(StaplerRequest req) throws IOException {
     }
 
     private void trackInitialPluginInstall(@Nonnull final List<Future<UpdateCenter.UpdateCenterJob>> installJobs) {
-        final Jenkins jenkins = Jenkins.getInstance();
+        final Jenkins jenkins = Jenkins.get();
         final UpdateCenter updateCenter = jenkins.getUpdateCenter();
         final Authentication currentAuth = Jenkins.getAuthentication();
 
-        if (!Jenkins.getInstance().getInstallState().isSetupComplete()) {
+        if (!Jenkins.getInstanceOrNull().getInstallState().isSetupComplete()) {
```
Comment (daniel-beck) [inline core/src/main/java/hudson/PluginManager.java]: Why? Would make the potential NPE explicit if it wouldn't already throw 4 lines up…?

Comment (oleg-nenashev) [reply core/src/main/java/hudson/PluginManager.java]: My suggestion was to keep using `jenkins` here. https://github.com/jenkinsci/jenkins/pull/4042#discussion_r287554871

Comment (GustavoBezerra) [reply core/src/main/java/hudson/PluginManager.java]: Sorry guys. I changed for use ```jenkins```instead

**Comment by GustavoBezerra** [comment]
PluginManager changed for use the variable ```jentkins```

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me now

**Review by res0nance — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Will merge it tomorrow if no negative feedback

File: core/src/main/resources/META-INF/upgrade/Jenkins.hint [hunk 10]
```diff
@@ -1,2 +1,2 @@
-jenkins.model.Jenkins.getInstance() => jenkins.model.Jenkins.get();;
+jenkins.model.Jenkins.get() => jenkins.model.Jenkins.get();;
```
Comment (jglick) [inline core/src/main/resources/META-INF/upgrade/Jenkins.hint]: Incorrect and fixed in #4112.

**Pull request merged**